### ISSUE TITLE
Native: tried calling BackgroundGeolocation.finish, but the BackgroundGeolocation plugin is not installed

### ISSUE THREAD

**Issue by avkonst** [ionic-team/ionic-native#3316]
**I'm submitting a ...**  (check one with "x")
[x ] bug report
[ ] feature request

**Current behavior:**
Capacitor+React enabled Ionic project warns about uninstalled background location plugin:
```
common.ts:330 Native: tried calling BackgroundGeolocation.finish, but the BackgroundGeolocation plugin is not installed
Install the BackgroundGeolocation plugin: 'ionic cordova plugin add @mauron85/cordova-plugin-background-geolocation'
```
When I run `ionic cordova plugin add @mauron85/cordova-plugin-background-geolocation` it says react can only work with capacitor, but not with cordova.

Note: `configure` method and `on` methods do not cause warnings and the plugin actually works and provides the location information. But calling `finish` causes the warning.

**Expected behavior:**
No warning in the development console.

**Steps to reproduce:**
1. Bootstrap new ionic app based on react
2. Add @ionic-native and @mauron85/cordova-plugin-background-geolocation packages
3. Render the following component in your app (see related code below)
4. See the warning in the dev console.

**Related code:**

```
import React from 'react'
import { isPlatform } from '@ionic/react';

import { BackgroundGeolocation, BackgroundGeolocationEvents, BackgroundGeolocationResponse, BackgroundGeolocationLocationProvider } from '@ionic-native/background-geolocation';

export const LocationTracker = () => {
    const [location, setLocation] = React.useState<BackgroundGeolocationResponse | undefined>(undefined)
    React.useEffect(() => {
        if (!isPlatform('hybrid')) {
            return () => {}
        }
        
        BackgroundGeolocation.configure({
            locationProvider: BackgroundGeolocationLocationProvider.RAW_PROVIDER,
            // desiredAccuracy: BackgroundGeolocationAccuracy.HIGH,
            stationaryRadius: 500,
            distanceFilter: 500,
            notificationTitle: 'Background tracking',
            notificationText: 'enabled',
            debug: true,
            interval: 10000,
            fastestInterval: 500,
            activitiesInterval: 10000,
        })

        BackgroundGeolocation.on(BackgroundGeolocationEvents.location).subscribe(
            (location: BackgroundGeolocationResponse) => {
                console.log(location)
                setLocation(location)
                BackgroundGeolocation.finish(); // FOR IOS ONLY, IMPORTANT!
            });


        BackgroundGeolocation.start();
        
        return () => {
            BackgroundGeolocation.removeAllListeners();
            BackgroundGeolocation.stop();
        }
    }, [])
    
    return <p>{location && location.speed}, {location && location.latitude}</p>;
}
```

**Other information:**
No other

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
Ionic:

   Ionic CLI       : 6.1.0 (/home/konstaan/.npm-global/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 5.0.0

Capacitor:

   Capacitor CLI   : 1.5.0
   @capacitor/core : 1.5.0

Utility:

   cordova-res : not installed
   native-run  : not installed

System:

   NodeJS : v10.19.0 (/usr/bin/node)
   npm    : 6.13.4
   OS     : Linux 4.15

```

**Comment by RaschidJFR** [comment]
Dup of #258

**Comment by aliexalter** [comment]
Same error tried every thing mention in #258 and ionic forum (ion-native).

**Comment by RaschidJFR** [comment]
I’m pretty sure the fix works. Let’s keep the conversation in the other thread. Could you please show your package.json content?

**Comment by aliexalter** [comment]
`{
  "name": "BackgroundTestApp",
  "version": "1.0.0",
  "author": "Ali Exalter",
  "homepage": "http://google.uk/",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~8.1.2",
    "@angular/core": "~8.1.2",
    "@angular/forms": "~8.1.2",
    "@angular/platform-browser": "~8.1.2",
    "@angular/platform-browser-dynamic": "~8.1.2",
    "@angular/router": "~8.1.2",
    "@ionic-native/background-geolocation": "^5.24.0",
    "@ionic-native/core": "^5.0.0",
    "@ionic-native/firebase": "^5.23.0",
    "@ionic-native/firebase-x": "^5.23.0",
    "@ionic-native/http": "^5.25.0",
    "@ionic-native/splash-screen": "^5.22.0",
    "@ionic-native/sqlite": "^5.22.0",
    "@ionic-native/status-bar": "^5.22.0",
    "@ionic/angular": "^5.0.0",
    "@mauron85/cordova-plugin-background-geolocation": "^3.1.0",
    "angular2-signaturepad": "^2.11.0",
    "angularfire2": "^5.4.2",
    "cordova-android": "^8.1.0",
    "cordova-ios": "^5.1.1",
    "cordova-plugin-advanced-http": "^2.4.1",
    "cordova-plugin-androidx": "^1.0.2",
    "cordova-plugin-androidx-adapter": "^1.1.0",
    "cordova-plugin-device": "^2.0.3",
    "cordova-plugin-file": "^6.0.2",
    "cordova-plugin-firebasex": "^9.0.2-cli",
    "cordova-plugin-ionic": "^5.4.6",
    "cordova-plugin-splashscreen": "^5.0.3",
    "cordova-plugin-statusbar": "^2.4.3",
    "cordova-plugin-whitelist": "^1.3.4",
    "cordova-sqlite-storage": "^5.0.0",
    "core-js": "^2.5.4",
    "firebase": "^7.14.0",
    "ion2-calendar": "^3.0.0-rc.0",
    "ionicsignaturepad": "^1.0.1",
    "moment": "^2.24.0",
    "ngx-order-pipe": "^2.0.4",
    "rxjs": "~6.5.1",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/architect": "~0.801.2",
    "@angular-devkit/build-angular": "~0.801.2",
    "@angular-devkit/core": "~8.1.2",
    "@angular-devkit/schematics": "~8.1.2",
    "@angular/cli": "^8.1.3",
    "@angular/compiler": "~8.1.2",
    "@angular/compiler-cli": "~8.1.2",
    "@angular/language-service": "~8.1.2",
    "@ionic/angular-toolkit": "^2.1.1",
    "@types/jasmine": "~3.3.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "^5.0.0",
    "cordova-plugin-ionic-keyboard": "^2.2.0",
    "cordova-plugin-ionic-webview": "^4.1.3",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.1.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.15.0",
    "typescript": "~3.4.3"
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-ionic-webview": {},
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-ionic": {
        "APP_ID": "3dde3072",
        "CHANNEL_NAME": "Master",
        "UPDATE_METHOD": "background",
        "MAX_STORE": "2",
        "MIN_BACKGROUND_DURATION": "30",
        "UPDATE_API": "https://api.ionicjs.com"
      },
      "cordova-sqlite-storage": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-device": {},
      "cordova-plugin-whitelist": {},
      "cordova-plugin-firebasex": {},
      "@mauron85/cordova-plugin-background-geolocation": {
        "GOOGLE_PLAY_SERVICES_VERSION": "11+",
        "ANDROID_SUPPORT_LIBRARY_VERSION": "26+",
        "ICON": "@mipmap/icon",
        "SMALL_ICON": "@mipmap/icon",
        "ALWAYS_USAGE_DESCRIPTION": "App requires background tracking ",
        "MOTION_USAGE_DESCRIPTION": "App requires motion detection"
      },
      "cordova-plugin-advanced-http": {
        "OKHTTP_VERSION": "3.10.0"
      }
    },
    "platforms": [
      "ios",
      "android"
    ]
  }
}`

**Comment by RaschidJFR** [comment]
Could you plz post it on the other thread?

**Comment by aliexalter** [comment]
I already posted.

**Comment by lrcampos97** [comment]
I have the same problem. 

Does anyone solve it?

**Comment by SergioZhydecky** [comment]
the same issue

**Comment by sushil50889** [comment]
same issue here

**Comment by RaschidJFR** [comment]
Yes, this have been solved in #258

**Comment by tobika** [comment]
For everybody who still has this issue (I guess only on iOS)
The problem is that the ionic-native declarations are not up to date with the plugin. `finish()` has been replaced with `endTask()`

I found the info here, unfortunately there was no update of the documentation/ionic-native for this yet.
https://github.com/ionic-team/ionic-native/issues/3360#issuecomment-606172675

Maybe it's better to use the plugin without ionic-native.
